session.go

```diff
@@ -73,6 +69,8 @@ type session struct {
 	// closeChan is used to notify the run loop that it should terminate.
 	// If the value is not nil, the error is sent as a CONNECTION_CLOSE.
 	closeChan chan *qerr.QuicError
+	// the error this session was closed with
+	closeErr  error
```

Comment:
For the error for closing a session in order to replace it with a new one (`errCloseSessionForNewVersion`), we're sending a `nil` into the channel. Same for the cases where we send a Public Reset instead of a ConnectionClose.